File: jinheung_session_auth/src/main/java/jinheung/project/user/repository/UserRepository.java
package jinheung.project.user.repository;


import jinheung.project.user.domain.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Long> {

    Long countByDeleted(Integer deleted);
    Optional<User> findByUsername(String username);

    int countBy();


}

File: jinheung_gateway/src/main/kotlin/jinheung.project/gateway/auth/r2dbc/repository/UserRepository.kt
package jinheung.project.gateway.auth.r2dbc.repository

import jinheung.project.gateway.auth.r2dbc.entity.User
import org.springframework.data.repository.kotlin.CoroutineCrudRepository
import javax.validation.constraints.Email

interface UserRepository : CoroutineCrudRepository<User, Long> {

}